datarepo-clienttests/src/main/java/common/utils/ProcessUtils.java

```diff
@@ -67,8 +67,23 @@ public static Process executeCommand(
    * @param proc the process handle
    * @return the list of lines written to stdout
    */
-  public static List<String> waitForTerminateAndReadStdout(Process proc) throws IOException {
-    return readStdout(proc, -1);
+  public static List<String> waitForTerminateAndReadStdout(Process proc)
```

Comment:
It seems like this revised method does 2 things:
1. waits for the process to terminate for a specified number of retries
2. tries to terminate it if it does not terminate in that time

- For (1), perhaps we could use Process.waitFor(long timeout, TimeUnit unit) instead? That way we don't have to write our own polling loop.
- For (2), there is another method in this class killProcessAndWaitForTermination(Process proc, long timeoutSeconds) that already handles this.

I'm not sure whether it's better to have the forced termination inside this method, or have it called from wherever is launching the process instead. This utility method is used in other places, so we should consider all current uses. Most of the time, I prefer smaller utility methods over ones that do lots of things, since it makes them more generally useful, but not sure whether that applies in this case. What do you think.